.pre-commit-config.yaml

```diff
@@ -0,0 +1,6 @@
+repos:
```

Comment:
file-level comment - as discussed online, I generally prefer pre-push linting over pre-commit linting; commits are okay to be messy, but pushes should be polished, imho. Any objection to pre-push instead?

Reply:
ahh I understand now, I actually have never experienced pre-push linting so I assumed you meant on CI and didn't know pre-push was an option.  But I get you, this way we can iterate faster on committing changes and only get slowed down by the linter when we attempt a push.